xcodeproj/internal/files.bzl

```
@@ -173,9 +173,9 @@ def normalized_file_path(file, *, folder_type_extensions):
     path = file.path
 
     for extension in folder_type_extensions:
-        prefix, ext, _ = path.partition(extension)
-        if not ext:
+        if extension not in path:
             continue
```

Is `endswith` slower because of the Starlark method lookup cost?
```suggestion
        if not path.endswith(extension):
            continue
```

---

I'd have to re-run the profile to make sure this doesn't slow things down, but given that it has better readability def worth re-running it. Will do and report back here.

---

I tried `find` too as a suggestion from @brentleyjones but it was `1s` slower than `partition` so not sure yet, will re-run and see how `.endswith` goes

---

That's why `partition` was being used to extract `prefix`, `ext` and leave out the rest

---

Ah, perfect. Didn't look at the implementation of partition here but figured it'd be at the end. Ignore my suggestion then!